standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java

```diff
@@ -1325,6 +1204,7 @@ private void validateTableType(Table tbl) {
     }
     validateTableType(tbl);
     sharedCache.addTableToCache(catName, dbName, tblName, tbl);
+    sharedCache.addTableConstraintsToCache(catName,dbName,tblName,new SQLAllTableConstraints());
```

Comment:
There is a separate api for createTableWithConstraints. 

Is this api also used when table is created with constraints?

Reply:
This line was added to update table cache with all constraints null and flag set to true. So avoid redirection to raw store.